### PR TITLE
Inline IANA data

### LINKED ISSUE #221: Proposal: Remove requests / requests cache / appdirs
Hello,

I think it it would be really great if this library could remove the `requests`, `requests-cache` and `appdirs` dependencies in it. The behavior of pulling down data in the the enums.py files ( see https://github.com/oasis-open/cti-stix-validator/blob/master/stix2validator/v21/enums.py#L2195 ) can lead to inputs which have differing validation statuses based on the operational environment in which this library code is executed in.

While pulling down and caching the IETF data locally is nice, this data could easily be added as package data, just like the schema .json files are packaged. This would remove an entire dependency chain from the library that isn't strictly necessary, while making the validation process more consistent across environments.

If a PR was put up for making this change, would it be considered?

### PR DESCRIPTION
**Pull request by vEpiphyte**
- Add script to download the IANA data.
- Inline all IANA data into the library into a per-version asset directory.
- Replace per-line processing of CSV files with csv library.
- Update enums files to read from the per-version assets.
- Remove requests, requests-cache, appdirs from core library.
- Remove caching options.

Closes #221

### CODE REVIEW

File: docs/contributing.rst [hunk 1]
```diff
@@ -130,3 +130,7 @@ must also be updated. To do so:
 
     git submodule update --remote
 
+Updating IANA cached data
+-------------------------
+
+TODO - Write documentation here.
```
Comment (vEpiphyte) [inline docs/contributing.rst:136]: open TODO

**Comment by vEpiphyte** [comment]
@clenk Thanks for your review.

I wasn't certain 100% how much was shared between the v20/v21 specification. If we assume those assets are the same, those can be merged down together.

**Comment by rpiazza** [comment]
@vEpiphyte -

The tests failed because the includes were not sorted correctly.  Please run "pre-commit run --all-files" and commit.

This will correct the order automatically.

**Comment by clenk** [comment]
@vEpiphyte I haven't done an exhaustive look at it, but I don't _think_ anything changed between the two versions. That's not to say it won't change in future STIX versions, but trying to account for that might be premature optimization.

**Pull request merged**